src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -115,14 +115,15 @@ public void run(ActionContext context) throws Exception {
                                                     config.isServiceAccountFilePath());
     BigQuery bigQuery = GCPUtils.getBigQuery(config.getProject(), credentials);
     //create dataset to store the results if not exists
-    String datasetName = config.getDataset();
-    String tableName = config.getTable();
-    String datasetProjectId = config.getDatasetProject();
     if (!Strings.isNullOrEmpty(datasetName) && !Strings.isNullOrEmpty(tableName)) {
       BigQuerySinkUtils.createDatasetIfNotExists(bigQuery, DatasetId.of(datasetProjectId, datasetName),
                                                  config.getLocation(), cmekKeyName,
                                                  () -> String.format("Unable to create BigQuery dataset '%s.%s'",
                                                                      datasetProjectId, datasetName));
+      if (cmekKeyName != null) {
```

Comment:
should this cmekKey to be set before createDatasets ? or it has not impact on datasets creation ?

Reply:
yeah, it has no impact on dataset creation, that follows a different codepath.